5-Cookies/server/tests/env/drivers/mongoDBDriver.ts

```
@@ -0,0 +1,30 @@
+import { MongoMemoryServer } from "mongodb-memory-server";
+
+import mongoose from "mongoose";
+import { errorHandler } from "../../../utils/errorHandler";
+import { MongoDB } from "../../../utils/MongoDB";
+
+export class MongoDriver extends MongoDB {
+  mongoServer;
+  contstrucor() {}
+  async setup() {
+    try {
+      this.mongoServer = await MongoMemoryServer.create();
+      await mongoose.connect(this.mongoServer.getUri());
+    } catch (error) {
+      errorHandler(error)
```

why swallowing errors?

---

Didn't understand how to act here, what should I do with these errors? and what does it mean swallowing errors?🙃

---

@ItayGershman - let's say that you failed to connect to the db. what do u want to happen? do u want to continue business as usual?

I guess not. But.. that's what this code says.. log it and continue as usual.
Then, when a request will need the DB it will obviously crash but it will be hard to know that it's not b/c of the request but it was b/c of the db that didn't manage to connect on startup.

We always want to fail fast. This way we will know fast that something is wrong and we can act accordingly. Also it will be easier to pin point the error.

in this specific case.. if the app fails to connect to the db it should try again and if it fails for the second time I would kill the server

---

Ok, now I get it. Now I try to reconnect only once, if it fails then I kill the server.